scripts/validate-schemas.js

```diff
@@ -1,90 +1,60 @@
 const path = require('path');
-
 const AjvDraft04 = require('ajv-draft-04');
 const ajvDraft04 = new AjvDraft04();
-
 const Ajv = require('ajv');
 const ajv = new Ajv();
 
-function validation (excludedFiles){
-
-  // Specify the path to the 'schemas' directory
-  const directoryPath = './schemas';
-
-  try{
+function validateSchema(filePath, fileContent, schemaValidator) {
+  try {
+    const obj = JSON.parse(fileContent);
+    const validate = schemaValidator(obj);
 
-    const files = fs.readdirSync(directoryPath);
+    if (validate) {
+      console.log(`\n${filePath}: JSON Schema is valid!`);
+    } else {
+      console.error(`\n${filePath}: JSON Schema is not valid:`, schemaValidator.errors);
+    }
 
-    // Filter files
-    const filteredFiles = files.filter(file =>  !excludedFiles.includes(file) && path.extname(file).toLowerCase() === '.json');
+    return validate;
+  } catch (error) {
+    console.error(`\n${filePath}: Error reading or parsing JSON Schema: ${error.message}`);
+    return false;
+  }
+}
 
+function validation(excludedFiles) {
+  const directoryPath = './schemas';
 
-    // Collect errors in an array
-    const validationErrors = [];
+  try {
+    const files = fs.readdirSync(directoryPath);
+    const filteredFiles = files.filter(file => !excludedFiles.includes(file) && path.extname(file).toLowerCase() === '.json');
     
-    // Iterate through the filtered files
+    const validationErrors = [];
+
     filteredFiles.forEach(file => {
-        // Construct the full path to the JSON schema file
-        const filePath = path.join(directoryPath, file);
-        
+      const filePath = path.join(directoryPath, file);
 
-        try {
-            // Read and parse the JSON schema
-            const fileContent = fs.readFileSync(filePath, 'utf8');
-            const obj = JSON.parse(fileContent);
-    
-            let validate;
-            if (obj.$schema === 'http://json-schema.org/draft-04/schema') {
-                // Validate the schema
-                validate = ajvDraft04.validateSchema(obj);
-                if(validate){
-                    console.log(`\n${file}: JSON Schema is valid!`);
-                }   
-            } else {
-                // Validate the schema
-                validate = ajv.validateSchema(obj);
-                if(validate){
-                    console.log(`\n${file}: JSON Schema is valid!`);
-                }
-            }
-    
-            // Check if the schema is not valid and collect errors
-            if (!validate) {
-                validationErrors.push({
-                    file,
-                    errors: obj.$schema === 'http://json-schema.org/draft-04/schema'
-                    ? ajvDraft04.errors
-                    : ajv.errors
-                });
-            }
-        } catch (error) {
-            validationErrors.push({
-            file,
-            errors: [{ message: `\nError reading or parsing JSON Schema: ${error.message}` }]
```

Comment:
We are missing this crucial part; the collection of errors

Reply:
> We are missing this crucial part; the collection of errors

@Min2who has this been implemented/discussed?